File: backend/src/main/java/com/mafia/dto/messages/MessageType.java
package com.mafia.dto.messages;

/**
 * All possible WebSocket message types.
 * Used as discriminator in WebSocketMessage envelope.
 * 
 * Keep in sync with frontend: src/types/websocket.ts
 */
public enum MessageType {
    // Lobby messages (broadcast to /topic/lobby/{id})
    PLAYER_LIST_UPDATE,    // Player joined/left/connected/disconnected
    PHASE_CHANGE,          // Game phase changed
    GAME_ANNOUNCEMENT,     // God made an announcement
    
    // Player messages (sent to /queue/player/{id})
    GAME_SNAPSHOT,         // Full game state on connect/reconnect
    ROLE_ASSIGNED,         // Your role has been assigned
    
    // Future message types
    // VOTE_UPDATE,        // Voting status changed
    // NIGHT_ACTION_RESULT,// Result of your night action
    // CHAT_MESSAGE,       // Chat message
    // PLAYER_ELIMINATED,  // A player was eliminated
    // GAME_ENDED,         // Game finished with winner
}

File: frontend/src/types/MessageType.ts
/**
 * WebSocket Message Type Enum
 *
 * Keep in sync with backend: com.mafia.dto.messages.MessageType
 */

export enum MessageType {
  // Lobby messages (broadcast to /topic/lobby/{id})
  PLAYER_LIST_UPDATE = "PLAYER_LIST_UPDATE",
  PHASE_CHANGE = "PHASE_CHANGE",
  GAME_ANNOUNCEMENT = "GAME_ANNOUNCEMENT",

  // Player messages (sent to /queue/player/{id})
  GAME_SNAPSHOT = "GAME_SNAPSHOT",
  ROLE_ASSIGNED = "ROLE_ASSIGNED",
}
